File: string/165.py
#数组越界这里python没有php中isset这种函数 只能try catch或者临时判断index跟length的关系。。。
class Solution:
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """

        l1 = version1.split(".")
        l2 = version2.split(".")

        length = len(l2) if len(l1) < len(l2) else len(l1)
        for i in range(0, length):
            try:
                a1 = int(l1[i]) if len(l1[i]) else 0
            except IndexError:
                a1 = 0
            try:
                a2 = int(l2[i]) if len(l2[i]) else 0
            except IndexError:
                a2 = 0
            if a1 < a2:
                return -1
            if a1 > a2:
                return 1
        return 0
        

File: two_pointers/165.cc
class Solution {
public:
    int compareVersion(string version1, string version2) {
        int i1 = 0, i2 = 0, v1 = 0, v2 = 0;
        while (i1 < version1.size() && i2 < version2.size()) {
            v1 = trans(i1, version1);
            v2 = trans(i2, version2);
            if (v1 < v2) return -1;
            if (v2 < v1) return 1;
        }
        while (i1 < version1.size()) {
            v1 = trans(i1, version1);
            if (v1 != 0) return 1;
        }
        while (i2 < version2.size()) {
            v2 = trans(i2, version2);
            if (v2 != 0) return -1;
        }
        return 0;
    }
private:
    int trans(int& index, std::string s) {
        std::string temp = "";
        while (index < s.size()) {
            if (s[index] == '.') {
                ++index;
                break;
            }
            temp += s[index];
            ++index;
        }
        if (temp == "") return 0;
        int ret = 0;
        for (const auto& c : temp) {
            ret = ret * 10 + c-'0';
        }
        return ret;
    }
};
